### PR TITLE
[consensus] replace infoMutex with atomic bool

### PR DESCRIPTION
**Pull request by LeoHChen**
Signed-off-by: Leo Chen <leo@harmony.one>

### CODE REVIEW

File: consensus/consensus_service.go [hunk 1]
```diff
@@ -160,9 +160,11 @@ func (consensus *Consensus) ResetState() {
 
 // ToggleConsensusCheck flip the flag of whether ignore viewID check during consensus process
 func (consensus *Consensus) ToggleConsensusCheck() {
-	consensus.infoMutex.Lock()
-	defer consensus.infoMutex.Unlock()
-	consensus.ignoreViewIDCheck = !consensus.ignoreViewIDCheck
+	if consensus.ignoreViewIDCheck.IsSet() {
```
Comment (JackyWYX) [inline consensus/consensus_service.go:163]: There is a potential read write race condition here. Please use atomic.Bool.Toggle. abool seems not supporting toggle implementation.

Comment (JackyWYX) [reply consensus/consensus_service.go:163]: Toggle implementation (from go-uber):

```
func (b *Bool) Toggle() bool {
	for {
		old := b.Load()
		if b.CAS(old, !old) {
			return old
		}
	}
}
```

Comment (LeoHChen) [reply consensus/consensus_service.go]: there is a fork with toggle support. forked it to our repo.

**Comment by LeoHChen** [comment]
@JackyWYX , can you review and approve this one?

**Comment by JackyWYX** [comment]
Where did you find the toggle implementation, there seems still to be a race condition here.

When you call `(*int32)(ab)`, it will create a new int32 and it has nothing to do with the original abool. If you use 

```
ab = ab.Toggle() 
``` 

in multiple threads, there could be read write race condition.

**Comment by JackyWYX** [comment]
Boolean variable is incrementing for each `Toggle` command, thus `IsSet` is broken. Please fix `IsSet` function.

Also I would recommend not to return value for the `Toggle` method, since there might be potential race condition if you call

```
ab = ab.Toggle()
```

**Comment by LeoHChen** [comment]
> Where did you find the toggle implementation, there seems still to be a race condition here.
> 
> When you call `(*int32)(ab)`, it will create a new int32 and it has nothing to do with the original abool. If you use
> 
> ```
> ab = ab.Toggle() 
> ```
> 
> in multiple threads, there could be read write race condition.

fair, I found the implementation in the original repo in a PR.

**Comment by JackyWYX** [comment]
> > Where did you find the toggle implementation, there seems still to be a race condition here.
> > When you call `(*int32)(ab)`, it will create a new int32 and it has nothing to do with the original abool. If you use
> > ```
> > ab = ab.Toggle() 
> > ```
> > 
> > 
> > in multiple threads, there could be read write race condition.
> 
> fair, I found the implementation in the original repo in a PR.

The deleted reply has some mistakes. Please refer to my last reply.

**Pull request merged**